pkg/plugins/tainttoleration/taint_toleration.go

```diff
@@ -38,52 +44,89 @@ func (pl *TaintToleration) Description() string {
 	if len(clusters) == 0 {
-		return clusters, nil
+		return plugins.PluginFilterResult{
+			Filtered: clusters,
+		}
 	}
 
 	// do validation on each toleration and return error if necessary
 	for _, toleration := range placement.Spec.Tolerations {
 		if len(toleration.Key) == 0 && toleration.Operator != clusterapiv1beta1.TolerationOpExists {
-			return nil, errors.New("If the key is empty, operator must be Exists.\n")
+			return plugins.PluginFilterResult{
+				Err: errors.New("If the key is empty, operator must be Exists.\n"),
+			}
 		}
 		if toleration.Operator == clusterapiv1beta1.TolerationOpExists && len(toleration.Value) > 0 {
-			return nil, errors.New("If the operator is Exists, the value should be empty.\n")
+			return plugins.PluginFilterResult{
+				Err: errors.New("If the operator is Exists, the value should be empty.\n"),
+			}
 		}
 	}
 
-	existingDecisions := getDecisions(pl.handle, placement)
+	decisionClusterNames := getDecisionClusterNames(pl.handle, placement)
 
 	// filter the clusters
 	matched := []*clusterapiv1.ManagedCluster{}
 	for _, cluster := range clusters {
-		if isClusterTolerated(cluster, placement.Spec.Tolerations, existingDecisions.Has(cluster.Name)) {
+		if isClusterTolerated(cluster, placement.Spec.Tolerations, decisionClusterNames.Has(cluster.Name), nil) {
 			matched = append(matched, cluster)
 		}
 	}
 
-	return matched, nil
+	return plugins.PluginFilterResult{
+		Filtered: matched,
+	}
+}
+
+func (pl *TaintToleration) RequeueAfter(ctx context.Context, placement *clusterapiv1beta1.Placement) plugins.PluginRequeueResult {
+	// get exist decisions clusters
+	decisionClusterNames, decisionClusters := getDecisionClusters(pl.handle, placement)
+	if decisionClusterNames == nil || decisionClusters == nil {
+		return plugins.PluginRequeueResult{}
+	}
+
+	requeueResults := &([]plugins.PluginRequeueResult{})
+	// filter and record pluginRequeueResults
+	for _, cluster := range decisionClusters {
+		isClusterTolerated(cluster, placement.Spec.Tolerations, decisionClusterNames.Has(cluster.Name), requeueResults)
+	}
+	// sort pluginRequeueResults and return the minimal RequeueTime
+	sort.SliceStable(*requeueResults, func(i, j int) bool {
+		t1 := (*requeueResults)[i].RequeueTime
+		t2 := (*requeueResults)[j].RequeueTime
+		return t1 != nil && t2 != nil && t1.Before(*t2)
+	})
+	if len(*requeueResults) > 0 {
+		return (*requeueResults)[0]
+	}
+
+	return plugins.PluginRequeueResult{}
 }
 
 // isClusterTolerated returns true if a cluster is tolerated by the given toleration array
-func isClusterTolerated(cluster *clusterapiv1.ManagedCluster, tolerations []clusterapiv1beta1.Toleration, inDecision bool) bool {
+func isClusterTolerated(cluster *clusterapiv1.ManagedCluster, tolerations []clusterapiv1beta1.Toleration, inDecision bool, requeueResult *[]plugins.PluginRequeueResult) bool {
```

Comment:
can we make 
```go
func isClusterTolerated(cluster *clusterapiv1.ManagedCluster, tolerations []clusterapiv1beta1.Toleration, inDecision bool) (bool, time.Duration)?
```

so it returns whether it is tolerated and how long it is tolerated.

Reply:
fixed.